lib/ansible/plugins/connection/local.py

```diff
@@ -132,6 +132,8 @@ def exec_command(self, cmd, in_data=None, sudoable=True):
 
     def _ensure_abs(self, path):
         if not os.path.isabs(path) and self.cwd is not None:
+            if isinstance(self.cwd, binary_type):
+                path = to_bytes(path, errors='surrogate_or_strict')
```

Comment:
When dealing with paths, it's usually best to ensure you are always dealing with bytes.  Instead of this conditional, the parts should always be ensured as bytes using `to_bytes`.  You may additionally need to validate that the callers of `_ensure_abs` always expect bytes.

Reply:
Thanks for the comments!

Fair point about checking what the caller wants. I changed to code to check the type of the input 'path' and make the return value correspond with that. The actual os.path.join will be done with bytes.